client/tx/legacy.go

```diff
@@ -6,28 +6,28 @@ import (
 	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
```

Comment:
Do you reckon we could move this file out the top-level packages in this PR? In the SDK we avoid make top-level packages depend on what is housed in `x/`.

I'd understand if you thought this is out of scope of this PR though :+1:

Reply:
This is not a top level package (it's in `client/tx` and it's already importing `github.com/cosmos/cosmos-sdk/x/auth/signing`).

I would need to move the whole file. If you think we should do it I can create a new issue.

Reply:
> This is not a top level package

Yes, you're right, please allow me rephrase it. The rule is: what does not live within the `x/*` directory structure must not depend on anything in `x/*` [1]. There are a couple of issues you can use as reference: https://github.com/cosmos/cosmos-sdk/issues/4488 https://github.com/cosmos/cosmos-sdk/issues/5724

> and it's already importing github.com/cosmos/cosmos-sdk/x/auth/signing.

Yes, I know that. You're just doing some work on this and that's great.

> I would need to move the whole file. If you think we should do it I can create a new issue.

It might require a bit of more refactoring, so a new issue would be greatly appreciated!

[1] We should really add a CI check to enforce this rule.

Reply:
No need to create new issue, the one you linked (#4488) already covers it - I've added one checkbox there.

Reply:
That was closed. We should open a new one IMHO

Reply:
Ah, I didn't notice it's closed. Created a new one: #7318